File: exchange-processing-service/src/main/java/io/kamenskiyAndrey/processingService/processing/model/AccountEvent.java
package io.kamenskiyAndrey.processingService.processing.model;

import lombok.Builder;
import lombok.Data;
import lombok.NonNull;

import java.math.BigDecimal;
import java.util.Date;

/*
Класс-модель события для отправки сообщения в Кафку
 */
@Data
@Builder
public class AccountEvent {
    @NonNull
    private String uuid; //уникальный номер операции

    private long userId, accountId;

    private Long fromAccount; //откуда списаны деньги

    @NonNull
    private String currencyCode; //код валюты

    @NonNull
    private Operation operation; // Операции - пополнение счета и перевод (тип операции)

    @NonNull
    private BigDecimal amount;

    @NonNull
    private Date created;
}

File: notification-bot/src/main/kotlin/io/kamenskiyAndrey/currencyExchanger/notification/model/AccountEvent.kt
package io.kamenskiyAndrey.currencyExchanger.notification.model

import java.math.BigDecimal
import java.util.*

/*
Класс который будет представлять десириализованное событие из кафки из JSON в класс. Создаем модель
описывающую JSON, который лежит в топике кафки.
 */
data class AccountEvent(
    val uuid: String,
    val accountId: Long,
    val userId: Long,
    val fromAccount: Long?,
    val currencyCode: String,
    val operation: Operation,
    val amount: BigDecimal,
    val created: Date
    )